roles/kubernetes/node/templates/kubelet.service.j2

```diff
@@ -24,6 +24,11 @@ ExecStart={{ bin_dir }}/kubelet \
 		$KUBELET_CLOUDPROVIDER
 Restart=always
 RestartSec=10s
+{% if kubelet_systemd_hardening %}
+# Hardening setup
+IPAddressDeny=any
+IPAddressAllow={{ groups['kube_control_plane'] | map('extract', hostvars, ['ansible_host']) | join(' ') }}
```

Comment:
We may be dealing with multi-homed hosts which have dedicated management interfaces (think ssh access only) and use a separate interface for kubelet traffic. This hardening setup should allow to bind to dedicated interfaces IMHO.

Reply:
What I meant was on this line:

```
{{ groups['kube_control_plane'] | map('extract', hostvars, ['ansible_host']) | join(' ') }}
```

You are making the assumption that only the main interface is used where the deployer may want to have the kubelet interface set to a different interface. `ansible_host` is the IP on which the ansible controller connects to the host which may not be the IP that the deployer wants to use for this type of traffic.

Reply:
Ok, now I got your point.
So, essentially we should list the IPs used by the `kube-apiserver` (**kubespray** variable `kube_apiserver_bind_address`), instead of the `ansible_host` variable.
This because the `kube-apiserver` is the only component that communicates with the `kubelet`.

Reply:
@cristicalin do you have suggestions on how to retrieve this information? I'm stuck here

Reply:
We could ask the deployer to supply the kubelet interface in the inventory variables, something like:

```yaml
kubelet_secure_interface: eth1
```

And then use `json_query` to extract the `ipv4.address` / `ipv6.address` fields.

Alternatively we can ask the deployer to specify explicit addresses for the nodes in the inventory:

```ini
[all]
kube-node-1 kubelet_secure_address=172.16.23.10
kube-node-2 kubelet_secure_address=172.16.23.11
```

Reply:
What if we set the `kubelet_secure_address` as `{{ groups['kube_control_plane']['hostvars']['ansible_host'] | join(' ') }}` by default, so the user should not deal with this variable in a common case. Instead, when needed, it can be changed.
In case the user bound the `kube-apiserver` to a different interface (that is not the one used by ansible), it should just define the variable in this way: `kubelet_secure_address="172.16.23.10,172.16.23.11,172.16.23.12"`.

Reply:
You should be mindful of variable precedence if doing that but I'd like to see an example of how the code would look like.

Reply:
I updated the code. Take a look and let me know what do you think :)

Reply:
The variable that you suggested is defined like so:
```yaml
 kubelet_secure_address: "{{ groups['kube_control_plane'] | map('extract', hostvars, ['ansible_host']) | join(' ') }}"
```

You can just override it in this way:
```yaml
kubelet_secure_address: "172.16.23.10 172.16.23.11 172.16.23.12"
```

Reply:
Sorry @cristicalin, did you take a look at the code?